src/solvers/thermal_operators.cpp

```diff
@@ -59,16 +60,13 @@ void DynamicConductionOperator::Mult(const mfem::Vector& u, mfem::Vector& du_dt)
 
   *bc_rhs_ = *rhs_;
   for (auto& bc : ess_bdr_) {
-    bc.eliminateToRHS(*K_mat_, y_, *bc_rhs_);
+    bc.eliminateToRHS(*K_mat_, u, *bc_rhs_);
   }
 
   // Compute:
   //    du_dt = M^{-1}*-K(u)
   // for du_dt
-  K_mat_->Mult(y_, z_);
-  z_.Neg();  // z = -zw z_.Add(1.0, *bc_rhs_);
-  z_.Add(1.0, *bc_rhs_);
-  M_solver_.Mult(z_, du_dt);
+  du_dt = M_solver_ * (-(*K_mat_ * y_) + *bc_rhs_);
```

Comment:
I think we might rename some of our EquationSolvers and matrices to be in more agreement with these overloaded operators,

e.g. substituting {`M_solver` -> `invM` and `K_mat` -> `K`} would make the code read

`du_dt = invM_ * (f_ - K_ * y_);`